scs: init at 2.0.2

bhipple opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

bhipple committed:
scs: init at 2.0.2

bhipple commented:
@GrahamcOfBorg build scs

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=b80a1796-74ae-4589-9db7-c03f00916065)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2
shrinking /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2/lib/libscsindir.so
shrinking /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2/lib/libscsdir.so
strip is /nix/store/4md2i310zklkkl5j41yw70gcwgn4kav5-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2/lib
patching script interpreter paths in /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2
checking for references to /build in /nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2...
/nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=312894a9-7db4-4bce-a769-75c1ebfd499f)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2
shrinking /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2/lib/libscsindir.so
shrinking /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2/lib/libscsdir.so
strip is /nix/store/553rihc190vsyy8b22iqcq25a6489h8y-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2/lib
/nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2
patching script interpreter paths in /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2
checking for references to /build in /nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2...
```
</p></details>

bhipple commented on pkgs/top-level/all-packages.nix:
```diff
@@ -20619,6 +20619,10 @@ with pkgs;
 
+  scs = callPackage ../development/libraries/science/math/scs {
+    liblapack = liblapackWithoutAtlas;
+  };
```
Presumably this would be fine with atlas, but I couldn't get atlas to build on my system.  It appears to be a source-only package so that it can do optimizations for your particular host CPU, so there are no Hydra binary caches and I figured ofborg probably wouldn't like it either.

bhipple commented on pkgs/development/libraries/science/math/scs/default.nix:
```diff
@@ -0,0 +1,40 @@
+  checkPhase = ''
+    ./out/demo_socp_indirect 42
+  '';
+
```
Upstream makefile doesn't provide standard check/install targets, but I think these get the job done.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=6dc52be4-7218-4f4e-b400-3e7d550cc7a9)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/laxb9jyrr6ij3h4amnml40y9j57hawxd-scs-2.0.2
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=eba34032-95c4-4464-98cf-555698701f62)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/7mm6dsl08yam49n1wy1lxa1svww6kzdn-scs-2.0.2
```
</p></details>

bhipple commented:
Oops, initial version didn't `cp -r include $out/`; fixed up now and added a comment or two about the test suite.  Should be good to go now:

```
$ tree $(nix-build . -A scs)
/nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2
├── include
│   ├── accel.h
│   ├── cones.h
│   ├── cs.h
│   ├── ctrlc.h
│   ├── glbopts.h
│   ├── linalg.h
│   ├── linsys.h
│   ├── normalize.h
│   ├── scs_blas.h
│   ├── scs.h
│   └── util.h
└── lib
    ├── libscsdir.a
    ├── libscsdir.so
    ├── libscsindir.a
    └── libscsindir.so

2 directories, 15 files
```

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=f0e353de-fa53-4fcd-97ab-5d5a2412d735)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2
shrinking /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2/lib/libscsdir.so
shrinking /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2/lib/libscsindir.so
strip is /nix/store/553rihc190vsyy8b22iqcq25a6489h8y-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2/lib
patching script interpreter paths in /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2
checking for references to /build in /nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2...
/nix/store/bdj7nyivwwq7s2liphk6v508di9c67zn-scs-2.0.2
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45294&attempt_id=4c42f1ab-2662-4a3f-a228-accd4039aa0f)

Attempted: scs

<details><summary>Partial log (click to expand)</summary><p>

```
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2
shrinking /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2/lib/libscsindir.so
shrinking /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2/lib/libscsdir.so
strip is /nix/store/4md2i310zklkkl5j41yw70gcwgn4kav5-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2/lib
patching script interpreter paths in /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2
checking for references to /build in /nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2...
/nix/store/z0zaakcqj4afk90qsb7ydxbi40r32ayb-scs-2.0.2
```
</p></details>
